GUI.ASM

```diff
@@ -47,6 +47,78 @@ proc pixel_plot
     ret
 endp
 
+proc draw_horizontal_line
+    arg @@x1:dword, @@x2:dword, @@y:dword, @@color:dword
+    uses eax, ebx, ecx, edx
+
+    mov eax, [@@x1]
+    mov ebx, [@@x2]
+
+    @@printline:
+        cmp eax, ebx
+        jg @@end
+        call pixel_plot, eax, [@@y], [@@color]
+        inc eax
+        jmp @@printline
+
+    @@end:
+    ret
+endp draw_horizontal_line
+
+proc draw_vertical_line
+    arg @@x:dword, @@y1:dword, @@y2:dword, @@color:dword
+
+    mov eax, [@@y1]
+    mov ebx, [@@y2]
+    @@printline:
+        cmp eax, ebx
+        jg @@end
+        call pixel_plot, [@@x], eax, [@@color]
+        inc eax
+        jmp @@printline
+
+    @@end:    
+    ret
+endp draw_vertical_line    
+
+proc draw_button
+    arg @@x1:dword, @@y1:dword, @@text:dword, @@backgroundcolor:dword
+
+    mov eax, [@@x1]
+    add eax, 200
```

Comment:
Maybe add the 200 and 80 as parameters to specify the size of the button so that we can differnet sized buttons. We can always add another wrapper function over that `draw_button_default` that intern just calls this function with 200 and 80.